server/src/main/java/org/elasticsearch/rollup/RollupShardDecider.java

```diff
@@ -0,0 +1,239 @@
+                                   IndexMetadata requestIndexMetadata,
+                                   SortedMap<String, IndexAbstraction> indexLookup) throws IOException {
+        IndexAbstraction originalIndex = indexLookup.get(requestIndexMetadata.getIndex().getName());
+        // Index must be member of a data stream
+        if (originalIndex.getParentDataStream() == null) {
+            return true;
+        }
+        // Rollup metadata must exist
+        Map<String, String> indexRollupMetadata = requestIndexMetadata.getCustomData(RollupIndexMetadata.TYPE);
+        if (indexRollupMetadata == null) {
+            return true;
+        }
+        // A rollup index is being searched
+        if (checkRollupConditions(request) == false) {
+            return false;
+        }
+
+        final AggregatorFactories.Builder aggregations = request.source() != null ? request.source().aggregations() : null;
+        final String rollupConfigSource = indexRollupMetadata.get(RollupIndexMetadata.ROLLUP_META_FIELD);
+        RollupIndexMetadata rollupIndexMetadata = RollupIndexMetadata.parseMetadataXContent(rollupConfigSource);
+
+        DateHistogramAggregationBuilder source = getDateHistogramAggregationBuilder(aggregations);
+        ZoneId sourceTimeZone = ZoneOffset.UTC; // Default timezone is UTC
+        if (source != null && source.timeZone() != null) {
+            sourceTimeZone = ZoneId.of(source.timeZone().toString(), ZoneId.SHORT_IDS);
+        }
+        DateHistogramInterval sourceInterval = source != null ? source.getCalendarInterval() : null;
+        // Incompatible timezone => skip this rollup group
+        if (canMatchTimezone(sourceTimeZone, rollupIndexMetadata.getDateTimezone().zoneId()) == false
+            || canMatchCalendarInterval(sourceInterval, rollupIndexMetadata.getDateInterval()) == false) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Parses the search request and checks if the required conditions are met so that results
+     * can be answered by a rollup index.
+     *
+     * Conditions that must be met are:
+     *  - Request size must be 0
+     *  - Metric aggregations must be supported
+     *  - No use of runtime fields in the query
+     *
+     * @param request the search request to parse
+     * @return true if a rollup index can
+     */
+    private static boolean checkRollupConditions(ShardSearchRequest request) {
+        if (request.source() == null) {
+            return false;
+        }
+
+        // If request size is not 0, rollup indices should not match
+        if (request.source().size() > 0) {
+            return false;
+        }
+
+        // Check for supported aggs
+        AggregatorFactories.Builder aggregations = request.source().aggregations();
+        if (checkSupportedAggregations(aggregations) == false){
+            return false;
+        }
+
+        // TODO(csoulios): Add check for runtime fields in the request
```

Comment:
We also need to inspect the search request and any unmapped field in aggs and queries.